OpenRA.Mods.RA/Traits/LeavesHusk.cs

```diff
@@ -41,18 +57,42 @@ public void Killed(Actor self, AttackInfo e)
 			if (!self.IsInWorld)
 				return;
 
+			if (self.World.SharedRandom.Next(100) > info.Probability)
+				return;
+
+			var warhead = e.Warhead as DamageWarhead;
+			if (info.DeathType != null && warhead != null && !warhead.DamageTypes.Contains(info.DeathType))
+				return;
+
 			self.World.AddFrameEndTask(w =>
 			{
 				var td = new TypeDictionary
 				{
 					new ParentActorInit(self),
 					new LocationInit(self.Location),
-					new CenterPositionInit(self.CenterPosition),
-					new OwnerInit(self.Owner),
-					new RaceInit(race),
-					new SkipMakeAnimsInit()
+					new CenterPositionInit(self.CenterPosition)
 				};
 
+				if (info.Owner == null)
+				{
+					td.Add(new OwnerInit(self.Owner));
+					td.Add(new RaceInit(race ?? self.Owner.Faction.InternalName));
+				}
+				else if (info.Owner == "Killer")
+				{
+					td.Add(new OwnerInit(e.Attacker.Owner));
+					td.Add(new RaceInit(race ?? e.Attacker.Owner.Faction.InternalName));
```

Comment:
You need to persist the original actor's race, otherwise you will break race-specific artwork.